streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -810,6 +813,36 @@ private StandbyTask findMatchingSuspendedStandbyTask(final TaskId taskId, final
         return null;
     }
 
+    private void closeNonAssignedSuspendedTasks() {
+        final Set<TaskId> currentActiveTaskIds = partitionAssignor.activeTasks().keySet();
+        final Iterator<Map.Entry<TaskId, StreamTask>> activeTaskIterator = suspendedActiveTasks.entrySet().iterator();
+        while (activeTaskIterator.hasNext()) {
+            closeAndRemoveIfNotAssigned(currentActiveTaskIds, activeTaskIterator.next(), activeTaskIterator);
```

Comment:
I'm not sure how returning a boolean helps with anything. It would just indicate that a single task has been removed. If we really want to make sure we've removed all the suspended tasks we can just check if the map is empty in `addStreamTasks` after the for loop.